### PR TITLE
[flask]: add /covidcast/correlation endpoint

### PR DESCRIPTION
**Pull request by sgratzl**
adds a new custom endpoint `/covidcast/correlation` which computes the correlation lag/r2 for a given date and window.

e.g.

http://localhost:5000/covidcast/correlation?reference=fb-survey:smoothed_cli&others=indicator-combination:confirmed_7dav_incidence_prop&geo=state:ny&window=20200101-20210402&lag=2

will result in: 

image

note: I just have limited local test data, so results may vary in your system

### CODE REVIEW

**Comment by chinandrew** [comment]
Is `Indicator` in signal source supposed to be capitalized?

**Comment by sgratzl** [comment]
> Is `Indicator` in signal source supposed to be capitalized?

that is just my local test system, I messed that up during my import of a sample dataset

File: src/server/endpoints/covidcast.py [hunk 1]
```diff
@@ -175,6 +178,64 @@ def gen(rows):
...
 
 
+@bp.route("/correlation", methods=("GET", "POST"))
+def handle_correlation():
+    require_all("reference", "window", "others", "geo")
+    reference = parse_single_source_signal_arg("reference")
+    other_pairs = parse_source_signal_arg("others")
+    geo_pairs = parse_geo_arg()
+    time_window = parse_day_range_arg("window")
+    lag = extract_integer("lag") or 28
```
[the remark this answers is not in the document]
Comment (krivard) [reply src/server/endpoints/covidcast.py:188]: (and would it be good to include these handlers in tests)

**Review by krivard — CHANGES_REQUESTED** [review]
Small truthy error, otherwise good

File: src/server/endpoints/covidcast.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sgratzl) [reply src/server/endpoints/covidcast.py]: good catch

**Review by krivard — APPROVED** [review]
nice work!

**Pull request merged**